### PR TITLE
Update SIMD developer docs.

### PR DESCRIPTION
**Pull request by halfflat**
SIMD class extensions and API modifications from the mechanism refactor work require updated developer documentation.

### CODE REVIEW

File: doc/simd_api.rst [hunk 1]
```diff
@@ -719,22 +801,46 @@ of type *J*, used only to disambiguate overloads.
...
+    :header-rows: 1
+    :widths: 20 20 60
+
+    * - Expression
+      - Type
+      - Description
+
+    * - ``C::cast_from(J{}, p, j)``
+      - ``C::vector_type``
+      - Vector *v* with values *v*\ `i`:sub: = *w*\ `i`:sub:, cast to ``C::scalar_type``.
```
Comment (noraabiakar) [inline doc/simd_api.rst]: The description isn't related to the arguments of cast_from. Also, why isn't J{} passed through tag?

File: doc/simd_api.rst [hunk 2]
```diff
@@ -514,6 +572,19 @@ In the following:
...
+      - Lane-wise cast to scalar type `L` as ``std::array``
+
+    * - ``simd_cast<simd<L, N, J>>(s)``
+      - ``simd<L, N, J>``
+      - Lane-wise cast to scalar type `L` as SIMD value.
+
+    * - ``simd_cast<S>(w)``
+      - *S*
+      - Lane-wise case to scalar type K from ``std::array`` as
+        SIMD value.
```
Comment (noraabiakar) [inline doc/simd_api.rst]: I'm not sure I understand the description, where does std::array come from?

**Comment by halfflat** [comment]
Thanks for the review! I'll revise and update.

File: doc/simd_api.rst [hunk 3]
```diff
@@ -463,8 +518,11 @@ In the following:
 
 * *A* is a SIMD class ``simd<K, N, I>`` for some scalar type *K*.
 * *S* is a SIMD class ``simd<V, N, I>`` for a floating point type *V*.
+* *W* is a SIMD class ``simd<L, N, J>`` for some scalar type *L* of the same width as *A*.
 * *a* and *b* are values of type *A*.
 * *s* and *t* are values of type *S*.
+* *w* is a value of type *W*.
+* *v* is a value of type ``std::array<W, N>``.
```
[the remark this answers is not in the document]
Comment (halfflat) [reply doc/simd_api.rst]: Yes, this was both wrong and incoherent. Will amend.

File: doc/simd_api.rst (hunk 1 above)
Comment (halfflat) [reply doc/simd_api.rst]: It definitely should have a `tag` attached.

**Comment by halfflat** [comment]
Hopefully this is now less wrong and slightly more readable.

**Pull request merged**